hack/generated/pkg/armclient/template_client.go

```diff
@@ -0,0 +1,322 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package armclient
+
+import (
+	"context"
+	"fmt"
+	"github.com/Azure/go-autorest/autorest/azure"
+	"github.com/Azure/go-autorest/autorest/azure/auth"
+	"github.com/Azure/k8s-infra/hack/generated/pkg/genruntime"
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"strings"
+)
+
+type (
+	// TODO: Naming?
+	Applier interface {
+		CreateDeployment(ctx context.Context, deployment *Deployment) (*Deployment, error)
+		DeleteDeployment(ctx context.Context, deploymentId string) error
+		GetDeployment(ctx context.Context, deploymentId string) (*Deployment, error)
+		NewDeployment(resourceGroup string, deploymentName string, resourceSpec genruntime.ArmResourceSpec) *Deployment
+
+		// TODO: Are we ok with the signature where the "result" is really a ptr to the last parameter?
```

Comment:
How is `error` a pointer to anything? 😕

Reply:
I phrased this poorly. This is saying: "The result of this function is a `status genruntime.ArmResourceStatus` which is currently constructed by the caller and filled out by this method"

I changed the text to:
```
		// TODO: These functions take an empty status and fill it out with the response from Azure (rather than as
		// TODO: the return type. I don't love that pattern but don't have a better one either.
```